hudi-common/src/main/java/org/apache/hudi/metadata/HoodieMetadataPayload.java

```diff
@@ -376,14 +377,27 @@ public Option<IndexedRecord> combineAndGetUpdateValue(IndexedRecord oldRecord, S
   }
 
   @Override
-  public Option<IndexedRecord> getInsertValue(Schema schemaIgnored, Properties propertiesIgnored) throws IOException {
+  public Option<IndexedRecord> getInsertValue(Schema schema, Properties propertiesIgnored) throws IOException {
     if (key == null || this.isDeletedRecord) {
       return Option.empty();
     }
 
-    HoodieMetadataRecord record = new HoodieMetadataRecord(key, type, filesystemMetadata, bloomFilterMetadata,
-        columnStatMetadata, recordIndexMetadata, secondaryIndexMetadata);
-    return Option.of(record);
+    if (schema != null && schema.getField(HoodieRecord.COMMIT_TIME_METADATA_FIELD) != null) {
+      GenericRecord result = new GenericData.Record(schema);
```

Comment:
not sure why we need this.

Reply:
* `HoodieMetadataPayload#getInsertValue` will be called as the merging result during FG reader merging metadata record.
* Currently `HoodieMetadataPayload#getInsertValue` always return fixed schema containing 7 data fields, which may break the logics in File Group reader, e.g., File Group reader has a required schema more than 7 data fields.
* The fix in this PR is using required schema to convert avro bytes back into GenericRecord.